examples/decoding/plot_receptive_field.py

```diff
@@ -0,0 +1,115 @@
+"""
+=========================================
+Receptive Field Estimation and Prediction
+=========================================
+
+This example re-creates figures from a paper from the Lalor group which
+demos the mTRF toolbox in Matlab [1]. We will show how the ReceptiveField
+class can perform a similar function along with scikit-learn. We'll fit
+a linear encoding model using the continuously-varying speech envelope to
+predict activity of a 128 channel EEG system.
+
+References
+----------
+[1] Crosse, M. J., Di Liberto, G. M., Bednar, A. & Lalor, E. C. (2016).
+The Multivariate Temporal Response Function (mTRF) Toolbox:
+A MATLAB Toolbox for Relating Neural Signals to Continuous Stimuli.
+Frontiers in Human Neuroscience 10, 604. doi:10.3389/fnhum.2016.00604
+"""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#
+# License: BSD (3-clause)
+
+
+from mne.decoding import ReceptiveField
+from sklearn.linear_model import Ridge
+from sklearn.model_selection import KFold
+from sklearn.preprocessing import scale
+
+import numpy as np
+import matplotlib.pyplot as plt
+import mne
+import scipy.io as si
+
+
+###############################################################################
+# Load the data from the publication
+path = mne.datasets.mtrf.data_path()
+data = si.loadmat(path + '/speech_data.mat')
+eeg = data['EEG'].T
+speech = data['envelope'].T
+sfreq = float(data['Fs'].squeeze())
+
+# Read in channel positions and create our MNE objects from the raw data
+mon = mne.channels.read_montage('biosemi128')
+mon.selection = mon.selection[:128]
+lt = mne.channels.read_layout('Vectorview-all')
+info = mne.create_info(mon.ch_names[:128], sfreq, 'eeg', montage=mon)
+lt = mne.channels.make_eeg_layout(info)
+eeg = mne.io.RawArray(eeg, info)
+
+###############################################################################
+# Plot a sample of brain and stimulus activity
+fig, ax = plt.subplots()
+_ = ax.plot(scale(eeg[:][0][..., :800].T), color='k', alpha=.1)
+ax.plot(scale(speech[0, :800]), color='r', lw=2)
+ax.set(title="Sample brain and stimulus activity", xlabel="Time (s)")
+
+###############################################################################
+# Create and fit a receptive field model
+
+# Define the lags that we'll use in the receptive field
+step_lag = .02
+lags = np.arange(.2, -.4, -step_lag)
+n_lags = len(lags)
+
+# Initialize the model
+spec_names = ['freq_%s' % ii for ii in range(speech.shape[0])]
+mod = Ridge()
+rf = ReceptiveField(lags, spec_names, sfreq=sfreq)
```

Comment:
ah I see - so you're thinking just always expect "lags" to be in samples. So users would need to convert lags (in seconds) to lags (in samples) on their own first? That sounds reasonable to me. I'll make that change if I'm correct in interpreting what you mention.